CHANGELOG.md

```diff
@@ -1,3 +1,128 @@
+New in 2.0
+==========
+
+* Many core kernel APIs have been redesigned and rewritten.
+  
+  * There are two new userspace system calls: `AllowReadOnly` and
+  `Exit`. The old `Allow` system call has been renamed to `AllowReadWrite`.
+  `AllowReadyOnly` provides a mechanism for userspace to share a 
+  read-only buffer (e.g., constant data stored in flash) to the kernel.
+  `Exit` allows a process to complete execution and request that the
+  kernel either terminate or restart it.
+  
+  * The system call ABI has been rewritten. System calls can now return
+  up to 4 registers of values to userspace on return. The ABI defines
+  the format and structure of the allowed return types. TRD104 documents
+  the new ABI.
+  
+  * The calling semantics and requirements for system calls is more
+  clearly defined, especially with respect to calls to the Allow
+  system calls and how buffers are managed. Furthermore, the lifetime
+  of upcalls passed to the kernel with the `subscribe` system call
+  has been defined. To enforce that the kernel doesn't maintain
+  references to upcalls that it shouldn't (so userspace can reclaim any
+  resources they require), upcalls are now managed by the kernel.
```

Comment:
"upcalls are now managed by the kernel" -> "upcalls are now managed by the core kernel"